test/jdk/java/awt/print/PrinterJob/SwingUIText.java

```diff
@@ -21,42 +21,70 @@
+ * @library /java/awt/regtesthelpers /test/lib
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
+ * @run main/manual SwingUIText
  */
 
-import java.awt.*;
-import javax.swing.*;
-import java.awt.print.*;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.GridLayout;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+import javax.swing.JButton;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import jtreg.SkippedException;
 
 public class SwingUIText implements Printable {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS = """
+            This test checks that when a Swing UI is printed,
+            the text in each component aligns with the component’s length as seen on-screen.
+            It also ensures the text spacing is reasonably even, though this is subjective.
+            The comparison should be made with JDK 1.5 GA or JDK 1.6 GA.
+
+            Steps:
+            1. Press the "Print" or "OK" button on the Print dialog.
+                This will print the content of the "Swing UI Text Printing Test" JFrame.
+            2. Compare the printout with the content of the JFrame.
+            3. If they match, press Pass; otherwise, press Fail.
+            """;
+
+    public static void main(String args[])  throws Exception {
+        PrinterJob job = PrinterJob.getPrinterJob();
+        if (job.getPrintService() == null) {
+            throw new SkippedException("Printer not configured or available.");
+        }
 
-    static String[] instructions = {
-        "This tests that when a Swing UI is printed, that the text",
-        "in each component properly matches the length of the component",
-        "as seen on-screen, and that the spacing of the text is of",
-        "reasonable even-ness. This latter part is very subjective and",
-        "the comparison has to be with JDK1.5 GA, or JDK 1.6 GA",
-    };
+        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .columns(45)
+                .testUI(SwingUIText::createTestUI)
+                .build();
 
-    static JFrame frame;
+        job.setPrintable(new SwingUIText());
+        if (job.printDialog()) {
+            job.print();
+        }
 
-    public static void main(String args[]) {
-        SwingUtilities.invokeLater(new Runnable() {
-          public void run() {
-              createUI();
-          }
-      });
+        passFailJFrame.awaitAndCheck();
     }
 
-    public static void createUI() {
-
-        Sysout.createDialogWithInstructions(instructions);
-
+    public static JFrame createTestUI() {
+        frame = new JFrame();
         JPanel panel = new JPanel();
         panel.setLayout(new GridLayout(4,1));
```

Comment:
```suggestion
        frame = new JFrame();
        JPanel panel = new JPanel();
        panel.setLayout(new GridLayout(4, 1));
```

Just because I happened to see it, you can add a space here to follow the precedent like in `displayText` just below this.